docs/automated-testing/unit-testing/mocking.md

```diff
@@ -92,16 +92,128 @@ since it is held to the same standard with full unit tests. Especially if writin
 developers can use, providing a fake in the product code means those developers no longer need to write their own mock
 implementations, further increasing re-usability of code.
 
-### Upsides
+### Upsides to Fakes
 
 - No framework needed, is just like any other implementation.
 - Encourages testable design.
 - Code can be "promoted" to product code, so it is not wasted effort.
 
-### Downsides
+### Downsides to Fakes
 
 - Takes more time to implement.
 
+## Best Practices
+
+To keep your mocking efficient, consider these best practices to make your code testable, save time and make your
```

Comment:
These best practices suggest mocking and DI are our preferred ways to do unit testing. 

IMHO we want to try to design away the need for mocking in general, and perhaps consider integration tests for the glue code that has outside dependencies, and avoid doing testing theatre where we add unit tests to get 100% on glue code that has little logic and that would be better covered in integration tests.

But then of course in C# and the likes, when we can't design away the need for unit tests of glue code for whatever reason, we want to make sure that we design for testing with DI etc.

Thoughts?

Reply:
That's a good point! The example I used in the Assertions best practice is very simple on purpose to convey the idea but avoid having overly complex code snippets. I agree that not every piece of glue code should be unit tested, but if a unit test with mocks is required, the principles still apply. I suggest adding a disclaimer to clarify better in which situations these principles should be used. What do you think? Or would you prefer changing the example to have more logic?

Reply:
Added. Let me know if this is okay, or you prefer further disclaimers or changes to the example :)